capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -464,4 +464,42 @@ else if (type.equals("java.util.List") || type.equals("java.util.ArrayList")) {
       return null;
     }
   }
+  
+   /**
+   * Deletes the object with the id given in the request query from the given collectionReference
+   * @param {collectionReference} reference to the appropriate collection of documents in the database
+   * @param {request} request sent to the backend
+   * @param {response} response returned from the call
+   * @return boolean true if the object is successfully deleted, false if not
+   */
+  public static boolean delete(CollectionReference collectionReference, HttpServletRequest request, 
+      HttpServletResponse response) throws IOException {
+    if (request.getParameterMap().size() > 1) {
+      System.err.println("Error: No other parameter can be sent with an ID");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return false;
+    }
+    if (request.getParameter("id") != null) {
+      String id = request.getParameter("id");
+      if (id.length() == 0) {
+        System.err.println("Error caused by an empty \"id\" field in the post body.");
+        response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+        return false;
+      }
+      ApiFuture<WriteResult> writeResult = collectionReference.document(id).delete();
+      try {
+        System.out.println("Update time : " + writeResult.get().getUpdateTime());
```

Comment:
I looked at this similar update code in Mauro's PR which he changed to have the name of the method, would it be good to have `"Delete Update Time :"` to be a little more specific? Not sure whether the backend logs provide the method signature

Reply:
That's a good idea, and could be helpful in debugging later on. I'll change these print statements in the other functions as well.